design/sales.md

```diff
@@ -135,70 +135,88 @@ the state is kept on local disk by the Repo and the Datastore. How much space is
+these requests should be removed as they are no longer fillable by the host.
+Note: expired request slots will be checked when a request is processed and its
+state is validated.
 
 ### Sort order
 
-Requests in the queue should be sorted in the following order:
-1. Profit (descending, yet to be determined how this will be calculated)
+Slots in the queue should be sorted in the following order:
+1. Profit (descending)<sup>1</sup>
 2. Collateral required (ascending)
 3. Time before expiry (descending)
 4. Dataset size (ascending)
 
+<sup>1</sup> While profit cannot yet be calculated correctly as this calculation will
+  involve bandwidth incentives, profit can be estimated as `duration * reward`
+  for now.
+
 Note: datset size may eventually be included in the profit algorithm and may not
-need to be included in the future. Additionally, data dispersal may also impact
-the datset size to be downloaded by the host, and consequently the profitability
-of servicing a storage request, which will need to be considered in the future
-once profitability can be calculated.
+need to be included on its own in the future. Additionally, data dispersal may
+also impact the datset size to be downloaded by the host, and consequently the
+profitability of servicing a storage request, which will need to be considered
+in the future once profitability can be calculated.
 
 ### Queue processing
 
 Queue processing will be started only once, when the sales module starts and
-will process requests continuously, in order, until the queue is empty. If the
+will process slots continuously, in order, until the queue is empty. If the
 queue is empty, processing of the queue will resume once items have been added
-to the queue. If the queue is not empty, but there are no availabilities, queue
-processing will resume once availabilites have been added.
-
-When a request is processed, it first is checked to ensure there is a matching
-availabilty, as these availabilities will have changed over time. Then, a random
-slot will be chosen, and the sales process will begin. If the request being
-processed contains one or more slot indices (as in the case of repair), the
-random slot should be chosen from that sample of indices. The start of the sales
-process should ensure that the random slot chosen is indeed available (slot
-state is "free") before continuing. If it is not available, the sales process
-will exit and the host will process the top request in the queue. The start of
-the sales process should also check to ensure the host is allowed to fill the
-slot, due to the [sliding window
+to the queue.
+
+As soon as items are available in the queue, and there are workers available for
+processing, an item is popped from the queue.
+
+> NOTE: queue processing should not be resumed once availabilites have been
+added as this would require keeping track of all requests on the network while
+there are no host availabilities, which would have a large memory impact.
```

Comment:
I don't understand this, could you elaborate?

Reply:
In the previous design, it was stated:
> If the queue is not empty, but there are no availabilities, queue processing will resume once availabilites have been added.

The issue with this is that if a host has no availabilities, then it must keep a separate list of all new storage requests coming in from the network, so that when availabilities are created by the host, the backlog of network requests can be compared to availabilities to potentially be added to the slot queue.

Reply:
I've updated the note to reflect this

Reply:
That makes sense, thanks for clarifying!